rabbitmqclient/cluster_reference_test.go

```diff
@@ -534,6 +533,142 @@ var _ = Describe("ParseReference", func() {
+							Name:      "rmq",
+							Namespace: namespace,
+						},
+					},
+				},
+			}
+			existingCredentialSecret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "rmq-default-user-credentials",
+					Namespace: namespace,
+				},
+				Data: map[string][]byte{
+					"username": []byte(existingRabbitMQUsername),
+					"password": []byte(existingRabbitMQPassword),
+				},
+			}
+			existingService = &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "rmq",
+					Namespace: namespace,
+				},
+				Spec: corev1.ServiceSpec{
+					ClusterIP: "1.2.3.4",
+					Ports: []corev1.ServicePort{
+						{
+							Name: "management",
+							Port: int32(15672),
+						},
+					},
+				},
+			}
+		})
+
+		When("requested namespace is prohibited", func() {
+			BeforeEach(func() {
+				existingRabbitMQCluster.ObjectMeta.Annotations = map[string]string{}
+				objs = []runtime.Object{existingRabbitMQCluster, existingCredentialSecret, existingService}
+			})
+			It("should return an error about a cluster being prohibited", func() {
+				_, _, err := rabbitmqclient.ParseReference(ctx, fakeClient,
+					topology.RabbitmqClusterReference{
+						Name:      existingRabbitMQCluster.Name,
+						Namespace: existingRabbitMQCluster.Namespace,
+					},
+					"prohibited-namespace",
+					"")
+				Expect(err).To(MatchError(rabbitmqclient.ResourceNotAllowedError))
+			})
+		})
+
+		When("there is a list of allowed namespaces", func() {
+			BeforeEach(func() {
+				existingRabbitMQCluster.ObjectMeta.Annotations = map[string]string{
+					"rabbitmq.com/topology-allowed-namespaces": "allowed1,allowed2",
+				}
+				objs = []runtime.Object{existingRabbitMQCluster, existingCredentialSecret, existingService}
+			})
+			When("requested namespace is in topology-allowed-namespaces annotation", func() {
+				It("works", func() {
+					_, _, err := rabbitmqclient.ParseReference(ctx, fakeClient,
+						topology.RabbitmqClusterReference{
+							Name:      existingRabbitMQCluster.Name,
+							Namespace: existingRabbitMQCluster.Namespace,
+						},
+						"allowed1",
+						"")
+					Expect(err).NotTo(HaveOccurred())
+
+					_, _, err = rabbitmqclient.ParseReference(ctx, fakeClient,
+						topology.RabbitmqClusterReference{
+							Name:      existingRabbitMQCluster.Name,
+							Namespace: existingRabbitMQCluster.Namespace,
+						},
+						"allowed2",
+						"")
+					Expect(err).NotTo(HaveOccurred())
+				})
+			})
+			When("requested namespace is not in topology-allowed-namespaces annotation", func() {
+				It("returns error", func() {
+					_, _, err := rabbitmqclient.ParseReference(ctx, fakeClient,
+						topology.RabbitmqClusterReference{
+							Name:      existingRabbitMQCluster.Name,
+							Namespace: existingRabbitMQCluster.Namespace,
+						},
+						"allowed3",
+						"")
+					Expect(err).To(MatchError(rabbitmqclient.ResourceNotAllowedError))
+				})
+			})
+		})
+
+		When("the rabbitmq allows all namespaces", func() {
```

Comment:
RabbitMQ does not allow the namespaces, the Operator does 🙃 may I suggest "when all namespaces are allowed" ?

Or if you want a funny tweak "when your allow list is made of stars" 🤪